adapters/src/main/java/io/realm/RealmBaseAdapter.java

```diff
@@ -87,10 +87,12 @@ public void onChange(BaseRealm results) {
     private void addListener(@NonNull OrderedRealmCollection<T> data) {
         if (data instanceof RealmResults) {
```

Comment:
check for instance of RealmCollectionObservable instead? and collapse the two if sections?